src/js/modules/infragistics.ui.editors.js

```diff
@@ -4595,6 +4595,19 @@
 			// have to perform this.options[ option ] = value;
 			$.Widget.prototype._setOption.apply(this, arguments);
 			switch (option) {
+				case "scientificFormat": {
+
+					//M.S. 3/16/2017 Issue 745 - When we set scientificFormat runtime, we cannot write 'e' or 'E' in edit mode.
+					if (prevValue) {
```

Comment:
Move this is statement into the other. Because you need it only if the new value is correct